shell/circleci/setup.sh

```diff
@@ -31,6 +31,13 @@ if [[ -n $TEST_RESULTS ]]; then
   mkdir -p "$TEST_RESULTS"
 fi
 
+# run prescript if user specified to install packages etc. before tests
+if [[ -n $PRE_SETUP_SCRIPT ]]; then
```

Comment:
This should also check if this is a file (or executable; see below), and error if it's not:
```suggestion
if [[ -n $PRE_SETUP_SCRIPT ]]; then
  if [[ ! -x $PRE_SETUP_SCRIPT ]]; then
    echo "PRE_SETUP_SCRIPT was not an executable; got "$PRE_SETUP_SCRIPT'"
    exit 1
  fi
```

Reply:
decided this is not needed